pkg/customunit/manager.go

```diff
@@ -0,0 +1,331 @@
+package customunit
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/Axway/agent-sdk/pkg/agent/cache"
+	"github.com/Axway/agent-sdk/pkg/amplify/agent/customunits"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	"github.com/Axway/agent-sdk/pkg/apic/apiserver/models/catalog/v1"
+	management "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	"github.com/Axway/agent-sdk/pkg/apic/definitions"
+	"github.com/Axway/agent-sdk/pkg/config"
+	"github.com/Axway/agent-sdk/pkg/transaction/models"
+	transUtil "github.com/Axway/agent-sdk/pkg/transaction/util"
+	"github.com/Axway/agent-sdk/pkg/util"
+	"github.com/Axway/agent-sdk/pkg/util/log"
+)
+
+type CustomUnitMetricServerManager struct {
+	configs          []config.MetricServiceConfiguration
+	cache            cache.Manager
```

Comment:
rather than the whole cache.Manager this should use a locally defined interface with just the methods in the cache that are used